compiler/resolution/typeSpecifier.cpp

```diff
@@ -45,6 +45,10 @@ Type* typeForTypeSpecifier(Expr* expr, bool fatalOK) {
                     int size = typeSize(var);
 
                     switch (size) {
+                    case 0:
+                      retval = dtBools[BOOL_SIZE_SYS];
+                      break;
```

Comment:
This and the other place returning / using a size of 0 just seem obviously wrong to me. I'd prefer another strategy for the system-sized bool but this PR is certainly making progress.